lib/pbench/agent/base.py

```diff
@@ -7,9 +16,89 @@ class BaseCommand(metaclass=abc.ABCMeta):
     def __init__(self, context):
         self.context = context
 
+        self.config = PbenchAgentConfig(self.context.config)
+
+        self.pbench_run = self.get_path(os.environ.get("pbench_run", None))
+        if self.pbench_run is None:
+            self.pbench_run = self.config.pbench_run
+        if not self.pbench_run.exists():
+            click.secho(
+                f"[ERROR] the provided pbench run directory, {self.pbench_run}, does not exist"
+            )
+            sys.exit(1)
+
+        # the pbench temporary directory is always relative to the $pbench_run
+        # directory
+        self.pbench_tmp = self.pbench_run / "tmp"
+        fs.safe_mkdir(self.pbench_tmp)
```

Comment:
Can we make `safe_mkdir` ensure the directory is either created or it raises an exception?

And it looks like it already does raise an exception which does not appear to be caught here.

Reply:
Done